src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueOfJsonPrimitive.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Text.Json.Nodes
+{
+    internal sealed class JsonValueOfJsonPrimitive : JsonValue
+    {
+        private readonly byte[] _value;
+        private readonly JsonValueKind _valueKind;
+        private readonly bool _isEscaped;
+
+        public JsonValueOfJsonPrimitive(ref Utf8JsonReader reader, JsonNodeOptions? options)
```

Comment:
Nit: a ref parameter in a constructor is somewhat unusual, a static factory might be preferable here.